probe-rs/src/probe/espusbjtag/protocol.rs

```diff
@@ -546,6 +596,113 @@ impl fmt::Debug for BitIter<'_> {
     }
 }
 
+use std::collections::VecDeque;
+
+/// An iterator over a received bit stream.
+#[derive(Clone)]
+pub struct OwnedBitIter {
```

Comment:
Don't we have a structure like this somewhere already?

Reply:
I couldn't find one, only the structs that work on borrowed slices.

Reply:
Yeah could be that it's only borrowed :)